src/cli.c

```diff
@@ -499,6 +499,51 @@ static void on_receive_datagram_frame(quicly_receive_datagram_frame_t *self, qui
         quicly_send_datagram_frames(conn, &payload, 1);
 }
 
+/// @brief Waut on UDP socket fd for read-readiness.
+/// @param fd UDP socket descriptor.
+/// @param connections array of connections to handle. The name must not conflict with global 'conns' var.
+/// @param num_connections The size of the connections array. The name must not conflict with global 'num_conns' var.
+/// @param client_enqueue_rq_at Pass the global 'enqueue_requests_at' var for a client. INT64_MAX for server.
+/// 
+/// @return a non-zero value if the UDP socket fd is read-ready; otherwise returns 0.
+static int udp_wait_readready(int fd, quicly_conn_t** connections, size_t num_connections, int64_t client_enqueue_rq_at)
+{
+    fd_set readfds;
+    struct timeval* tv, tvbuf;
+    do {
+        int64_t timeout_at = INT64_MAX;
+        for (size_t i = 0; i != num_connections; ++i) {
+            int64_t conn_to = quicly_get_first_timeout(connections[i]);
+            if (conn_to < timeout_at)
+                timeout_at = conn_to;
+        }
+
+        // client enqueues a new request to send.
+        if (client_enqueue_rq_at < timeout_at)
+            timeout_at = client_enqueue_rq_at;
+
+        if (timeout_at != INT64_MAX) {
+            int64_t delta = timeout_at - ctx.now->cb(ctx.now);
```

Comment:
Here in case of client the connection context was used instead of the global `ctx`. Does it make any difference?
```c
quicly_context_t *ctx = quicly_get_context(conn);
```